File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/grpc/CustomMetadataFileMessageBuilder.java
/**------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *------------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.grpc;

import java.util.function.Consumer;

import com.ericsson.adp.mgmt.data.BackupData;
import com.ericsson.adp.mgmt.data.CustomMetadataFileChunk;
import com.ericsson.adp.mgmt.data.CustomMetadataFileChunk.Builder;
import com.ericsson.adp.mgmt.data.DataMessageType;
import com.google.protobuf.ByteString;

/**
 * Builds messages to send custom metadata file through Data Channel.
 */
public class CustomMetadataFileMessageBuilder implements BackupMessageBuilder {

    @Override
    public BackupData getFileNameMessage(final String fileName) {
        return generateMessage(customMetadataFileChunkBuilder -> customMetadataFileChunkBuilder.setFileName(fileName));
    }

    @Override
    public BackupData getDataMessage(final ByteString data) {
        return generateMessage(customMetadataFileChunkBuilder -> customMetadataFileChunkBuilder.setContent(data));
    }

    @Override
    public BackupData getChecksumMessage(final String checksum) {
        return generateMessage(customMetadataFileChunkBuilder -> customMetadataFileChunkBuilder.setChecksum(checksum));
    }

    private BackupData generateMessage(final Consumer<Builder> setMessageInformationFunction) {
        final Builder messageBuilder = CustomMetadataFileChunk.newBuilder();
        setMessageInformationFunction.accept(messageBuilder);
        return wrapMessage(messageBuilder.build());
    }

    private BackupData wrapMessage(final CustomMetadataFileChunk chunk) {
        return BackupData
                .newBuilder()
                .setCustomMetadataFileChunk(chunk)
                .setDataMessageType(DataMessageType.CUSTOM_METADATA_FILE)
                .build();
    }

}

File: bro-agent-api/src/main/python/bro_agent/grpc/CustomMetadataFileMessageBuilder.py
from bro_agent.grpc.BackupMessageBuilder import BackupMessageBuilder
import bro_agent.generated.INT_BR_ORCH_DATA_pb2 as INT_BR_ORCH_DATA


class CustomMetadataFileMessageBuilder(BackupMessageBuilder):
    """
    * Builds messages to send custom metadata file through Data Channel.
    """
    @staticmethod
    def get_file_name_message(file_name):
        """ get the file name """
        backup_file_chunk = INT_BR_ORCH_DATA.CustomMetadataFileChunk(fileName=file_name)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.CUSTOM_METADATA_FILE,
                                                  customMetadataFileChunk=backup_file_chunk)
        return backup_data

    @staticmethod
    def get_data_message(data):
        """ get data message """
        backup_file_chunk = INT_BR_ORCH_DATA.CustomMetadataFileChunk(content=data)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.CUSTOM_METADATA_FILE,
                                                  customMetadataFileChunk=backup_file_chunk)
        return backup_data

    @staticmethod
    def get_checksum_message(checksum):
        """"get checksum message """
        backup_file_chunk = INT_BR_ORCH_DATA.CustomMetadataFileChunk(checksum=checksum)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.CUSTOM_METADATA_FILE,
                                                  customMetadataFileChunk=backup_file_chunk)
        return backup_data
